src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/StreamExtensions.cs

```diff
@@ -78,6 +78,9 @@ internal static int TryReadAll(this Stream stream, byte[] buffer, int offset, in
 
 #if NETCOREAPP
         internal static int TryReadAll(this Stream stream, Span<byte> buffer)
+#if NET7_0_OR_GREATER
```

Comment:
For my own education: Why is this specific case guarded by this preprocessor directive?

Reply:
I waffled on whether I should change this code or not. All the rest of the changes are to code that only ship in-box in the shared framework. `System.Reflection.Metadata` also ships in an OOB package, thus it builds for `net6.0` and `net7.0`. These new Stream APIs are only available in `net7.0`, and this code needs to handle both. So it needs to take separate paths.